snap/info.go

```diff
@@ -596,6 +596,21 @@ func BadInterfacesSummary(snapInfo *Info) string {
 	return strings.TrimSuffix(buf.String(), "; ")
 }
 
+// DesktopPrefix returns the prefix string for the desktop files that
+// belongs to the given snapInstance. We need to do something custom
+// here because a) we need to be compatible with the world before we had
+// parallel installs b) we can't just use the usual "_" parallel installs
+// separator because that is already used as the separator between snap
+// and desktop filename.
+func (s *Info) DesktopPrefix() string {
```

Comment:
I'm slightly wary of moving this info the "snap" package - I had hoped we could keep it an implementation detail of the desktop file writing. But given the apparmor need to know about this it seems that is unavoidable.

Reply:
We already have `AppInfo.DesktopFile()` in this package, and it was returning the wrong thing for parallel installed snaps prior to these changes.

Reply:
it would be nice if we didn't need this on Info, and could somehow do the addition of "+" from the AppInfo, but given the fact that an AppInfo doesn't know if it belongs to a parallel snap or not, we do need something at the snap level.